autosklearn/pipeline/components/data_preprocessing/text_encoding/tfidf_encoding.py

```diff
@@ -50,26 +51,20 @@ def fit(self, X: PIPELINE_DATA_DTYPE, y: Optional[PIPELINE_DATA_DTYPE] = None
             else:
                 raise KeyError()
 
-            all_text = itertools.chain.from_iterable(X[col].dropna() for col in X.columns)
+            all_text = itertools.chain.from_iterable(X[col] for col in X.columns)
             self.preprocessor = self.preprocessor.fit(all_text)
 
         else:
             raise ValueError("Your text data is not encoded in a pandas.DataFrame\n"
                              "Please make sure to use a pandas.DataFrame and ensure"
-                             "that the text features are encoded as strings.")
+                             " that the text features are encoded as strings.")
         return self
 
     def transform(self, X: PIPELINE_DATA_DTYPE) -> PIPELINE_DATA_DTYPE:
-        X_new = None
+        X.fillna("", inplace=True)
         if self.preprocessor is None:
             raise NotImplementedError()
-        # iterate over the pretrained preprocessors and columns and transform the data
-        for feature in X.columns:
-            if X_new is None:
-                X_new = self.preprocessor.transform(X[feature])
-            else:
-                X_new += self.preprocessor.transform(X[feature])
-        return X_new
+        return sum(self.preprocessor.transform(X[feature]) for feature in X.columns)
```

Comment:
Hey,  did you check the return type of `sum`? I would image in just sums up everything and does not care about dimensions, etc.